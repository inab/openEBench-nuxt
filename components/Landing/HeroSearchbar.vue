<template>
	<section height="100" dark class="primary lighten-1">
		<v-container class="pt-7 pb-7">
			<v-text-field
				v-model="input"
				:prepend-inner-icon="'mdi-magnify'"
				label="Search for individual tools"
				solo
				clearable
				hide-details
				data-testid="input-field"
				@keyup.enter="handleSubmit"
				@click:prepend-inner="handleSubmit"
			></v-text-field>
		</v-container>
	</section>
</template>

<script>
export default {
	name: 'HeroSearchbar',
	data: () => ({
		input: '',
	}),
	methods: {
		handleSubmit() {
			this.$router.push({
				path: '/tool',
				query: { search: this.input },
			});
		},
	},
};
</script>
