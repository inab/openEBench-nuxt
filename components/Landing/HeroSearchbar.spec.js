import { mount } from '@vue/test-utils';
import HeroSearchbar from './HeroSearchbar.vue';

const factory = () => {
	return mount(HeroSearchbar, {
		...createComponentMocks({}),
		mocks: {
			$router: { push: jest.fn() },
		},
	});
};

describe('HeroSearchbar', () => {
	it('is instantiated', () => {
		const wrapper = factory();
		expect(wrapper).toBeTruthy();
	});

	it('should match snapshot', () => {
		const wrapper = factory();
		expect(wrapper.html()).toMatchSnapshot();
	});

	it('should push router to tools page on button click', () => {
		const wrapper = factory();

		const submitBtn = wrapper.find('button');
		expect(submitBtn.exists()).toBe(true);
		submitBtn.trigger('click');

		expect(wrapper.vm.$router.push).toHaveBeenCalledWith({
			path: '/tool',
			query: { search: '' },
		});
	});

	it('should push router to tools page on enter press in input field', async () => {
		const wrapper = factory();

		const searchInput = wrapper.find('[data-testid="input-field"]');
		expect(searchInput.exists()).toBe(true);

		await searchInput.trigger('keyup.enter');

		expect(wrapper.vm.$router.push).toHaveBeenCalledWith({
			path: '/tool',
			query: { search: '' },
		});
	});

	it('should push router to tools with query on button click', () => {
		const wrapper = factory();

		const searchInput = wrapper.find('[data-testid="input-field"]');
		expect(searchInput.exists()).toBe(true);
		searchInput.setValue('Random Tool Name');

		const submitBtn = wrapper.find('button');
		expect(submitBtn.exists()).toBe(true);

		submitBtn.trigger('click');

		expect(wrapper.vm.$router.push).toHaveBeenCalledWith({
			path: '/tool',
			query: { search: 'Random Tool Name' },
		});
	});
});
