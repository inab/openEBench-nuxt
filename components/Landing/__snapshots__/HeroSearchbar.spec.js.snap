// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroSearchbar should match snapshot 1`] = `
"<section height=\\"100\\" dark=\\"\\" class=\\"primary lighten-1\\">
  <div class=\\"container pt-7 pb-7\\">
    <div class=\\"v-input v-input--hide-details theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed\\">
      <div class=\\"v-input__control\\">
        <div class=\\"v-input__slot\\">
          <div class=\\"v-input__prepend-inner\\">
            <div class=\\"v-input__icon v-input__icon--prepend-inner\\"><button type=\\"button\\" aria-label=\\"prepend icon\\" class=\\"v-icon notranslate v-icon--link mdi mdi-magnify theme--light\\"></button></div>
          </div>
          <div class=\\"v-text-field__slot\\"><label for=\\"input-6\\" class=\\"v-label theme--light\\" style=\\"left: 0px; position: absolute;\\">Search for individual tools</label><input data-testid=\\"input-field\\" id=\\"input-6\\" type=\\"text\\"></div>
          <div class=\\"v-input__append-inner\\">
            <div></div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>"
`;
